src/frontend/src/binder/expr/mod.rs

```diff
@@ -302,6 +304,75 @@ impl Binder {
         Ok(FunctionCall::new(func_type, vec![expr])?.into())
     }
 
+    pub(super) fn bind_distinct_from(
+        &mut self,
+        negated: bool,
+        left: Expr,
+        right: Expr,
+    ) -> Result<ExprImpl> {
+        let left = self.bind_expr(left)?;
+        let right = self.bind_expr(right)?;
+
+        let func_call = if negated {
+            // x IS NOT DISTINCT FROM y is equivalent to (x IS NULL AND y IS NULL) OR x = y
+            FunctionCall::new_unchecked(
+                ExprType::Or,
+                vec![
+                    FunctionCall::new_unchecked(
```

Comment:
Is there any preference for `new_unchecked` over `new` or the other way around? @xiangjinwu

Reply:
As the name and doc suggested, `new_unchecked` bypasses type checking and should only be used for a good reason.
Here the innermost expressions (IsNull, IsNotNull, Equal, NotEqual) are using `new` correctly, and outer expressions (And, Or) are bypassing the checks because both operands are from validated inner expressions. This is acceptable, but maybe the performance gain is negligible.

Reply:
Btw each clone of expr means repeated evaluation in the backend. It may be better to implement these 2 operators in backend directly rather than rewriting to such long expressions in the frontend.

Reply:
I see, let this PR add some tests first to make sure the semantics are all correct. 

Then another PR implements the operators in the backend and removes the rewriting.